src/main/java/co/cask/plugin/format/DBTypes.java

```diff
@@ -99,7 +103,11 @@
       case Types.DATE:
       case Types.TIME:
       case Types.TIMESTAMP:
-        type = Schema.Type.LONG;
+        if (convertDate) {
+          type = Schema.Type.STRING;
```

Comment:
will be good to have this logic under case `DATE`. else we will return type as string for timestamp too if convertDate is set to true.

Reply:
nvm, i noticed time and timestamp are also transformed to date below.